### PR TITLE
Filter on workflow for upload status logic

### PR DESCRIPTION
**Pull request by seallard**
Small cleanup, more efficient to filter out any non-fastq analyses earlier on since those are the only analyses we want to set the upload date for automatically currently.

### CODE REVIEW

File: trailblazer/services/analysis_service/utils.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ def create_update_analyses_response(analyses: list[Analysis]) -> UpdateAnalysesR
 
 def get_upload_date(analysis: Analysis) -> datetime | None:
     completed_job: Job | None = _get_completed_job(analysis.upload_jobs)
```
Comment (islean) [inline trailblazer/services/analysis_service/utils.py:49]: Not for this PR to solve, but I it is slightly annoying that analysis.upload_jobs is typehinted as singular here. Is it plural or singular?

**Review by islean — APPROVED** [review]
LGTM!

File: trailblazer/store/crud/read.py [hunk 2]
```diff
@@ -253,8 +253,13 @@ def _paginate_analyses(self, analyses: Query, request: AnalysesRequest) -> Query
...
         return apply_analysis_filter(
-            filter_functions=[AnalysisFilter.BY_NOT_UPLOADED, AnalysisFilter.BY_COMPLETED],
+            filter_functions=[
+                AnalysisFilter.BY_NOT_UPLOADED,
+                AnalysisFilter.BY_COMPLETED,
+                AnalysisFilter.BY_WORKFLOW,
+            ],
             analyses=self.get_query(Analysis),
+            workflow=Workflow.FASTQ,
         ).all()
```
Comment (ChrOertlin) [inline trailblazer/store/crud/read.py:265]: can we make this a general function that receives a workflow?

Comment (seallard) [reply trailblazer/store/crud/read.py]: Sure 👍 https://github.com/Clinical-Genomics/trailblazer/pull/469/commits/8a0a8b125d3e38518604b8bc6879b8c6f453951b

File: trailblazer/services/analysis_service/utils.py (hunk 1 above)
Comment (seallard) [reply trailblazer/services/analysis_service/utils.py:49]: It is plural, but the function just retrieves one of the completed upload jobs.

Usually there is only one upload job, but there can also be several failed ones for FASTQ

**Pull request merged**